consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -81,9 +87,15 @@ impl<T: Payload> ChainedBftSMR<T> {
                 let pre_select_instant = Instant::now();
                 let idle_duration;
                 select! {
-                    _ = reconfig_events.select_next_some() => {
+                    payload = reconfig_events.select_next_some() => {
                         idle_duration = pre_select_instant.elapsed();
-                        epoch_manager.start_processor().await
+                        let epoch_info = EpochInfo {
+                            epoch: payload.epoch(),
+                            verifier: Arc::new((&payload.get::<ValidatorSet>()
```

Comment:
why both performing this arc, why not just have a copy?

Reply:
epoch_info is sharing the verifier iirc so just best practice to avoid copies